### PR TITLE
QE-903: create service for activate and deactivate survey

### PR DESCRIPTION
**Pull request by lajosarpad**
As per https://limesurvey.atlassian.net/browse/QE-903, this is the server-side of that task

### CODE REVIEW

File: application/models/services/SurveyActivate.php [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+                $survey->{$field} = App()->request->getPost($field, $params[$field] ?? null);
+            }
+            $survey->save();
+
+            // Make sure the saved values will be picked up
+            Survey::model()->resetCache();
+            $survey->setOptions();
+        }
+
+        $surveyActivator = new SurveyActivator($survey);
```
Comment (kevin-foster-uk) [inline application/models/services/SurveyActivate.php:67]: `SurveyActivator` should be a constructor dependency.

File: application/models/services/SurveyActivate.php [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+            $fields = [
+                'anonymized',
+                'datestamp',
+                'ipaddr',
+                'ipanonymize',
+                'refurl',
+                'savetimings'
+            ];
+            foreach ($fields as $field) {
+                $survey->{$field} = App()->request->getPost($field, $params[$field] ?? null);
```
Comment (kevin-foster-uk) [inline application/models/services/SurveyActivate.php:58]: App() can be injected using LSYii_Application

File: application/models/services/SurveyActivate.php [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+            }
+            $survey->save();
+
+            // Make sure the saved values will be picked up
+            Survey::model()->resetCache();
```
Comment (kevin-foster-uk) [inline application/models/services/SurveyActivate.php:63]: Ideally we should inject `Survey` also but I am not sure if calling `resetCache()` on the injected instance will have the desired effect. In that case its fine to leave this as is.

File: application/models/services/SurveyActivate.php (hunk 1 above)
Comment (lajosarpad) [reply application/models/services/SurveyActivate.php]: @kevin-foster-uk applied the change accordingly

File: application/models/SurveyActivator.php
[the remark this answers is not in the document]
Comment (lajosarpad) [reply application/models/SurveyActivator.php:1]: @kevin-foster-uk it was not my choice, see https://github.com/LimeSurvey/LimeSurvey/blame/a99ecb718e3fe053eb8293940394183853365d76/application/models/SurveyActivator.php#L7

[the remark this answers is not in the document]
Comment (kevin-foster-uk) [reply application/models/SurveyActivator.php:1]: Ah I see those models already existed. That explains it.

**Review by kevin-foster-uk — APPROVED** [review]

File: application/models/SurveyDeactivator.php [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+
+    /** @var array */
+    protected $timingsTableDefinition = [];
+
+    /** @var string */
+    protected $error;
+
+    public function __construct($survey = null)
+    {
+        $this->$survey = $survey;
```
Comment (TonisOrmisson) [inline application/models/SurveyDeactivator.php:24]: a bug here, should be $this->s.., not $this->$s... Error 500 "Creation of dynamic property SurveyDeactivator::$ is deprecated"

Comment (lajosarpad) [reply application/models/SurveyDeactivator.php:24]: Thanks, @TonisOrmisson for pointing this out, created a PR to fix it

**Pull request merged**